src/cls/hello/cls_hello.cc

```diff
@@ -258,6 +258,35 @@ static int bad_writer(cls_method_context_t hctx, bufferlist *in, bufferlist *out
 }
 
 
+class PGLSHelloFilter : public PGLSFilter {
+  string val;
+public:
+  PGLSHelloFilter(bufferlist::iterator& params) {
+    ::decode(xattr, params);
+    ::decode(val, params);
```

Comment:
We should try to catch exceptions anywhere a decode is happening from stuff coming from a client.  Let's do that here as an example for future filter writers.